### PR TITLE
Fix: flex reparent/insert target indicators

### PR DESCRIPTION
**Pull request by balazsbajorics**
**Problem:**
The flex reparent/insert target indicators are attached to the flex sibling element. This is because the indicators were actually just a placeholder implementation while I worked on the reparent, but then in lieu of more time, I merged them to master.

**Fix:**
The indicators now respect the flex gap property.

**Commit Details:**
- Added flex gap property to specialSizeMeasurements
- Using the measured flex gap to offset the indicators

**Note:** 
- The indicators still don't have the size of the dragged element
- if the computed style's flex gap is somehow different from the visual one, they will fail

Here's how it looks like:
https://screenshot.click/05-08-rugm8-5xq16.mp4

### CODE REVIEW

File: editor/src/components/canvas/canvas-strategies/reparent-strategy-helpers.ts [hunk 1]
```diff
@@ -723,16 +738,16 @@ export function applyFlexReparent(
               const targetLineBeginningOfParent: CanvasRectangle =
                 newParentFlexDirection === 'row'
                   ? canvasRectangle({
-                      x: parentRect.x,
+                      x: parentRect.x, // TODO offset the insert line by half the flex gap?
```
Comment (seanparsons) [inline editor/src/components/canvas/canvas-strategies/reparent-strategy-helpers.ts:741]: Not sure if this TODO needs doing now.

File: editor/src/components/canvas/dom-walker.ts [hunk 2]
```diff
@@ -893,6 +893,13 @@ function getSpecialMeasurements(
     height: globalFrame.height - border.top - border.bottom,
   })
 
+  // TODO is there a nicer way to get the number only if it's in pixels, and otherwise set it to null?
```
Comment (seanparsons) [inline editor/src/components/canvas/dom-walker.ts:896]: You could `flatMapEither` it and then `defaultEither` it, but it's arguable which is more comprehensible.

**Review by seanparsons — APPROVED** [review]

**Comment by enidemi** [comment]
Based on flex-gap it's possible that the insertion indicator will be outside of the flex area.

Screenshot 2022-09-06 at 11 21 51

**Comment by balazsbajorics** [comment]
closing it in favor of #2518

**Pull request closed without merging**